testsuite/MDAnalysisTests/coordinates/test_netcdf.py

```diff
@@ -195,6 +201,393 @@ def test_dt(self, u):
+         [16.03358, 16.183628, 14.02995]]
+    ], dtype=np.float32)
+
+    frc_refs = np.array([
+        [[8.583388, 1.8023694, -15.0033455],
+         [-21.594835, 39.09166, 6.567963],
+         [4.363016, -12.135163, 4.4775457]],
+        [[-10.106646, -7.870829, -10.385734],
+         [7.23599, -12.366022, -9.106191],
+         [-4.637955, 11.597565, -6.463743]]
+    ], dtype=np.float32)
+
+    vel_refs = np.array([
+        [[-0.5301689, -0.16311595, -0.31390688],
+         [0.00188578, 0.02513031, -0.2687525],
+         [0.84072256, 0.09402391, -0.7457009]],
+        [[-1.7773226, 1.2307, 0.50276583],
+         [-0.13532305, 0.1355039, -0.05567304],
+         [-0.6182481, 1.6396415, 0.46686798]]
+    ], dtype=np.float32)
+
+    box_refs = np.array([
+        [28.81876287, 28.27875261, 27.72616397, 90., 90., 90.],
+        [27.06266081, 26.55555665, 26.03664058, 90., 90., 90.]
+    ], dtype=np.float32)
+
+    @pytest.fixture(scope='class')
+    def universe(self):
+        return mda.Universe(PRM_NCBOX, TRJ_NCBOX)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_positions(self, universe, index, expected):
+        universe.trajectory[index]
+        assert_almost_equal(self.coord_refs[expected],
+                            universe.atoms.positions[:3], self.prec)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_forces(self, universe, index, expected):
+        """Here we multiply the forces by 4.184 to convert from
+        kcal to kj
+        """
+        universe.trajectory[index]
+        assert_almost_equal(self.frc_refs[expected] * 4.184,
+                            universe.atoms.forces[:3], self.prec)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_velocities(self, universe, index, expected):
+        """Here we multiply the velocities by 20.455 to match the value of
+        `scale_factor`
+        """
+        universe.trajectory[index]
+        assert_almost_equal(self.vel_refs[expected] * 20.455,
+                            universe.atoms.velocities[:3], self.prec)
+
+    @pytest.mark.parametrize('index,expected', ((0, 1.0), (8, 9.0)))
+    def test_time(self, universe, index, expected):
+        assert_almost_equal(expected, universe.trajectory[index].time,
+                            self.prec)
+
+    def test_nframes(self, universe):
+        assert_equal(10, universe.trajectory.n_frames)
+
+    def test_dt(self, universe):
+        ref = 1.0
+        assert_almost_equal(ref, universe.trajectory.dt, self.prec)
+        assert_almost_equal(ref, universe.trajectory.ts.dt, self.prec)
+
+    @pytest.mark.parametrize('index,expected', ((0, 0), (8, 1)))
+    def test_box(self, universe, index, expected):
+        universe.trajectory[index]
+        assert_almost_equal(self.box_refs[expected], universe.dimensions)
+
+
+class _NCDFGenerator(object):
+    """A class for generating abitrary ncdf files and exhaustively test
```

Comment:
Interesting.

Perhaps [hypothesis](https://hypothesis.readthedocs.io/en/latest/) (which we are already using elsewhere) might also be of interest?

Reply:
So I had a look at hypothesis. I'll be honest, in part because I don't think I fully understand the API, I have no idea how best to employ hypothesis instead (or if it's applicable to this scenario). Any pointers here would be greatly appreciated.

Reply:
@tylerjereddy used it – maybe he has some pointers. For this PR I am totally happy with your tests. I just thought that learning new tools that might make life easier is not a bad thing.